plugins/frontend/backstage-plugin-github-insights/src/apis/GithubClient.ts

```diff
@@ -96,27 +101,28 @@ export class GithubClient implements GithubApi {
         continue;
       }
 
-      const linkLowerCased = mediaLink.toLocaleLowerCase('en-US');
-      const isAbsolute = linkLowerCased.startsWith('http');
-
-      let url;
-      // 1. Media link is relative to the readme file - eg. './img/myimage.png'
-      if (!isAbsolute) {
-        url = new URL(mediaLink, response.data.url);
-      } else if (linkLowerCased.includes(resource)) {
-        // 2. Media link is not referencing to an external github instance
-        // eg. https://github.com/myorg/myrepo/blob/master/32x32/gcp-logs.png
-        // we need a relative reference to create GitHub API call
-        const {
-          owner: ownerLink,
-          name: repoLink,
-          ref: refLink,
-          filepath,
-        } = parseGitUrl(mediaLink);
-
-        url = `/repos/${ownerLink}/${repoLink}/contents/${filepath}?ref=${refLink}`;
-      }
+      const getUrl = () => {
+        const linkLowerCased = mediaLink.toLocaleLowerCase('en-US');
+        if (!linkLowerCased.startsWith('http')) {
+          return new URL(mediaLink, response.data.url);
+        }
+        if (linkLowerCased.includes('github')) {
```

Comment:
This means we load in content via data url for most cases except relative links, which means we can render images in private repos correctly